NullReferenceException by finished process

### Description

tracing a running .net core 5.0 process. can't share the code
After the process is exited by itselft, the dotnet-trace ends and produces an error message quoted below

```
> dotnet trace collect --process-id 99999
...
[00:00:10:23]	Recording trace 393,4774 (MB)
Press <Enter> or <Ctrl+C> to exit...

Trace completed.
[ERROR] System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Diagnostics.Tools.Trace.CollectCommandHandler.Collect(CancellationToken ct, IConsole console, Int32 processId, FileInfo output, UInt32 buffersize, String providers, String profile, TraceFileFormat format, TimeSpan duration, String clrevents, String clreventlevel, String name, String diagnosticPort, Boolean showchildio) in /_/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs:line 315
```

### Configuration

macos big sur, intel
.net sdk 5.0.203
dotnet-trace 5.0.221401

Just happened to me as well. Latest dotnet-trace global tool (5.0.221401), targeting .NET 6 Preview 5.21226.1 process. After process exit tracing fails with:

```console
Trace completed.
[ERROR] System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.Diagnostics.Tools.Trace.CollectCommandHandler.Collect(CancellationToken ct, IConsole console, Int32 processId, FileInfo output, UInt32 buffersize, String providers, String profile, TraceFileFormat format, TimeSpan duration, String clrevents, String clreventlevel, String name, String diagnosticPort, Boolean showchildio) in /_/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs:line 315
```

On Windows 10.

--- a/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
+++ b/src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs
@@ -312,7 +312,7 @@ namespace Microsoft.Diagnostics.Tools.Trace
                         // If the process is shutting down by itself print the return code from the process.
                         // Capture this before leaving the using, as the Dispose of the DiagnosticsClientHolder
                         // may terminate the target process causing it to have the wrong error code
-                        if (ProcessLauncher.Launcher.ChildProc.WaitForExit(5000))
+                        if (ProcessLauncher.Launcher.HasChildProc && ProcessLauncher.Launcher.ChildProc.WaitForExit(5000))
                         {
                             ret = ProcessLauncher.Launcher.ChildProc.ExitCode;
                             Console.WriteLine($"Process exited with code '{ret}'.");

